appinsights/datacontracts.go

```diff
@@ -138,19 +156,29 @@ func (item *MetricTelemetry) baseData() Domain {
 	return item.data
 }
 
+func (item *MetricTelemetry) SetProperty(key, value string) {
+	if item.data.Properties == nil {
+		item.data.Properties = make(map[string]string)
+	}
+	item.data.Properties[key] = value
+}
+
 type RequestTelemetry struct {
 	BaseTelemetry
 	data *requestData
 }
 
-func NewRequestTelemetry(name string, timestamp time.Time, duration time.Duration, responseCode string, success bool) *RequestTelemetry {
+func NewRequestTelemetry(name, httpMethod, url string, timestamp time.Time, duration time.Duration, responseCode string, success bool) *RequestTelemetry {
 	now := time.Now()
```

Comment:
I wonder if it would be better to have a layer of indirection between `RequestTelemetry` - which holds the data to be placed in the envelope, and the envelope schema itself